lib/discovery_model/analyse_path.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// TODO(Move this to the driver code)
+
+/// Utility for extracting the features from a root folder
+library smart.discovery_models.analysis_path;
+
+import 'package:cli_util/cli_util.dart' as cli_util;
+
+import 'dart:io' as io;
+import 'dart:async';
+import 'feature_extractor.dart' as extractor;
+import 'package:sintr_common/logging_utils.dart' as log;
+
+String pathToSdk;
+
+Future<Map> analyseFolder(String path) async {
```

Comment:
I'm not sure about the code conventions in dart (or this project) but a comment on this method would be nice :)